### PR TITLE
fix: Duplicating arrow without bound elements throws error

### LINKED ISSUE #8315: Duplicate bound arrow without it's bound elements throws error
Create a rectangle and bind one end of an arrow. 
ALT+drag to duplicate arrow.

The arrow is not duplicated but error thrown in `fixBindingsAfterDuplication`
The underlying issue seems to be:
image

https://github.com/user-attachments/assets/cd92cb97-d3cb-4902-82d4-faa2a1afd8e3

### PR DESCRIPTION
**Pull request by zsviczian**
Resolves #8315, introduced in #8185

### CODE REVIEW

File: packages/excalidraw/element/binding.ts [hunk 1]
```diff
@@ -1246,6 +1246,9 @@ export const fixBindingsAfterDuplication = (
     .filter(({ id }) => allBindableElementIds.has(id))
     .forEach((bindableElement) => {
       const oldElementId = duplicateIdToOldId.get(bindableElement.id);
+      if (!oldElementId) {
+        return; //if no bound element is found
+      }
```
Comment (dwelle) [inline packages/excalidraw/element/binding.ts:1251]: typescript is doing us really dirty here and I don't know why

Comment (dwelle) [reply packages/excalidraw/element/binding.ts:1251]: actually nvm, I just noticed the assertion below

Comment (dwelle) [reply packages/excalidraw/element/binding.ts:1251]: can you pls remove that assertion as well?

Comment (zsviczian) [reply packages/excalidraw/element/binding.ts:1251]: so now do you want me to change anything or not?

Comment (dwelle) [reply packages/excalidraw/element/binding.ts:1251]: by assertion I meant on the `find()` call as it's still unsafe like this

File: packages/excalidraw/element/binding.ts [hunk 2]
```diff
@@ -1246,11 +1246,10 @@ export const fixBindingsAfterDuplication = (
     .filter(({ id }) => allBindableElementIds.has(id))
     .forEach((bindableElement) => {
       const oldElementId = duplicateIdToOldId.get(bindableElement.id);
-      const { boundElements } = sceneElements.find(
-        ({ id }) => id === oldElementId,
-      )!;
+      const { boundElements } =
+        sceneElements.find(({ id }) => id === oldElementId) ?? {};
```
Comment (dwelle) [inline packages/excalidraw/element/binding.ts:1250]:
```suggestion
      const boundElements =
        sceneElements.find(({ id }) => id === oldElementId)?.boundElements;
```

cleaner

**Comment by dwelle** [comment]
Alright, thanks for the fix! ❤️

**Review by dwelle — APPROVED** [review]

**Pull request merged**